include/boost/geometry/algorithms/simplify.hpp

```diff
@@ -624,6 +641,35 @@ struct simplify_multi
 };
 
 
+template <typename Geometry>
+struct has_same_tag_as
+{
+    template <typename OtherGeometry>
+    struct pred
+        : std::is_same
+            <
+                typename geometry::tag<Geometry>::type,
+                typename geometry::tag<OtherGeometry>::type
+            >
+    {};
+};
+
+template <typename StaticGeometryIn, typename DynamicGeometryOut>
+struct static_geometry_type
+{
+    using type = typename util::sequence_find_if
+        <
+            typename traits::geometry_types<DynamicGeometryOut>::type,
+            detail::simplify::has_same_tag_as<StaticGeometryIn>::template pred
+        >::type;
+
+    BOOST_GEOMETRY_STATIC_ASSERT(
```

Comment:
We keep using this? Or would new code use just `static_assert` ?

Reply:
For consistency and also because this macro has variadic number of arguments [1] so you can pass types after condition and message and these types will typically be displayed in the error message. Furthermore if you pass `false` as condition it will compile as long as a struct inside is not instantiated while `static_assert` will never compile. This is why we can't use `static_assert(false)` in all cases, e.g. inside `not_implemented`.

[1] https://github.com/boostorg/geometry/blob/develop/include/boost/geometry/core/static_assert.hpp#L24